### PR TITLE
add list row entries for expenses

### PR DESCRIPTION
**Pull request by brendansudol**
related to https://github.com/18F/cms-hitech-apd/issues/938, adds clickable list entries for expenses so you can see the relevant info at a glance and/or edit when needed.

### This pull request is ready to merge when...
- [x] Tests have been updated (and all tests are passing)
- [x] This code has been reviewed by someone other than the original author

### This feature is done when...
- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

**Comment by brendansudol** [comment]
Preview:
https://cl.ly/0f032N470e23

File: web/src/containers/activity/Expenses.js [hunk 1]
```diff
@@ -24,26 +26,172 @@ const EXPENSE_CATEGORIES = [
...
+  handleChange: PropTypes.func.isRequired,
+  handleYearChange: PropTypes.func.isRequired,
+  handleDelete: PropTypes.func.isRequired
+};
+
+const getLastExpense = expenses =>
+  expenses.length ? expenses[expenses.length - 1].key : null;
+
 class Expenses extends Component {
+  static getDerivedStateFromProps(props, state) {
```
Comment (mgwalker) [inline web/src/containers/activity/Expenses.js]: It looks like this is never used, but it'd almost satisfy [Laura's comment](https://github.com/18F/cms-hitech-apd/pull/955#pullrequestreview-148177788).  It looks like this code is intended to default the **last** expense to opened.

**Comment by mgwalker** [comment]
@lauraponce Would it make sense to get rid of the remove button at the bottom of the expense form in favor of the `X` button at the far right of the expense header?

**Comment by lauraponce** [comment]
@mgwalker totally! That would work perfectly (putting an X in the top right and getting rid of the remove button at the bottom).

File: web/src/containers/activity/Expenses.js (hunk 1 above)
Comment (brendansudol) [reply web/src/containers/activity/Expenses.js]: this is used -- when a new expense gets added, it adds that new expense key to the state and defaults it to be open

**Comment by brendansudol** [comment]
changes coming up!

**Comment by brendansudol** [comment]
updates made @lauraponce :)

**Comment by lauraponce** [comment]
@brendansudol one last small thing — when you click "Add expense", can it collapse the previous entry into the list format? I could see people just continuing to add more without clicking to collapse so they wouldn't get the cool benefit of the high-level view list.

**Comment by brendansudol** [comment]
good call @lauraponce -- i just made that change and the experience is a lot nicer with it :)

**Pull request merged**